src/Microsoft.AspNetCore.Session/DistributedSession.cs

```diff
@@ -210,6 +261,12 @@ private void Deserialize(Stream content)
                 int dataLength = DeserializeNumFrom4Bytes(content);
                 _store[key] = ReadBytes(content, dataLength);
             }
+
+            if (_logger.IsEnabled(LogLevel.Debug))
+            {
+                _sessionId = new Guid(_sessionIdBytes).ToString();
```

Comment:
It's just a question of lazy coverting the byte to string. The individual session keys are similarly lazy.